htdocs/compta/facture/card.php

```diff
@@ -2050,7 +2050,7 @@
 					$extrafields->fetch_name_optionals_label($nextSituationInvoice->table_element);
 					$ret = $extrafields->setOptionalsFromPost(null, $nextSituationInvoice);
 					if ($ret > 0) {
-						$nextSituationInvoice->insertExtraFields();
+						$nextSituationInvoice->insertExtraFields('BILL_CREATE');
```

Comment:
Problem here is that the trigger BILL_CREATE was already called into createFromCurrent.
I suggest to try this. Just before the 
`$object->createFromCurrent()`
do
```
$extrafields->fetch_name_optionals_label($$object->table_element);
$ret = $extrafields->setOptionalsFromPost(null, $object);
```
so the extrafields are already set when calling the createFromCurrent.Does this work for you ?